pina/solvers/pinn.py

```diff
@@ -121,7 +129,13 @@ def training_step(self, batch, batch_idx):
 
             # PINN loss: equation evaluated on location or input_points
             if hasattr(condition, 'equation'):
-                target = condition.equation.residual(samples, self.forward(samples))
+                if isinstance(condition.equation, ParametricEquation):
+                    target = condition.equation.residual(samples,
+                            self.forward(samples),
+                            self.problem.unknown_parameters)
+                else:
+                    target = condition.equation.residual(samples,
+                        self.forward(samples))
                 loss = self.loss(torch.zeros_like(target), target)
```

Comment:
what about if we pass to Equation params as extra argument? @ndem0 in this way we avoid having two classes and a lot of `if isistance(equation, ParametricEquation)`.... In a user perspective which wants to make a solver, remembering all this stuff (equation, parametric equation, ...) is not so easy.